File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/loading/LoadingFragment.java
package com.jdroid.android.sample.ui.loading;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;

import com.jdroid.android.activity.ActivityLauncher;
import com.jdroid.android.fragment.AbstractFragment;
import com.jdroid.android.sample.R;

public class LoadingFragment extends AbstractFragment {

	@Override
	public Integer getContentFragmentLayout() {
		return R.layout.loading_fragment;
	}

	@Override
	public void onViewCreated(View view, Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);

		findView(R.id.blockingLoading).setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {
				ActivityLauncher.startActivity(getActivity(), new Intent(getActivity(), BlockingLoadingActivity.class));
			}
		});
		findView(R.id.customBlockingLoading).setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {
				ActivityLauncher.startActivity(getActivity(), new Intent(getActivity(), CustomActivityLoadingActivity.class));
			}
		});
		findView(R.id.nonBlockingLoading).setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {
				ActivityLauncher.startActivity(getActivity(), new Intent(getActivity(), NonBlockingLoadingActivity.class));
			}
		});
		findView(R.id.swipeRefresLoading).setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {
				ActivityLauncher.startActivity(getActivity(), new Intent(getActivity(), SwipeRefreshLoadingActivity.class));
			}
		});
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/loading/LoadingFragment.kt
package com.jdroid.android.sample.ui.loading

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.jdroid.android.activity.ActivityLauncher
import com.jdroid.android.fragment.AbstractFragment
import com.jdroid.android.sample.R

class LoadingFragment : AbstractFragment() {

    override fun getContentFragmentLayout(): Int? {
        return R.layout.loading_fragment
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        findView<View>(R.id.blockingLoading).setOnClickListener {
            ActivityLauncher.startActivity(
                activity,
                Intent(activity, BlockingLoadingActivity::class.java)
            )
        }
        findView<View>(R.id.customBlockingLoading).setOnClickListener {
            ActivityLauncher.startActivity(
                activity,
                Intent(activity, CustomActivityLoadingActivity::class.java)
            )
        }
        findView<View>(R.id.nonBlockingLoading).setOnClickListener {
            ActivityLauncher.startActivity(
                activity,
                Intent(activity, NonBlockingLoadingActivity::class.java)
            )
        }
        findView<View>(R.id.swipeRefresLoading).setOnClickListener {
            ActivityLauncher.startActivity(
                activity,
                Intent(activity, SwipeRefreshLoadingActivity::class.java)
            )
        }
    }
}
